cli/azd/pkg/project/scaffold_gen.go

```diff
@@ -204,21 +255,187 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
+					userResourceName, usedResourceName, usedResourceName)
+			}
+			var err error
+			switch usedResource.Type {
+			case ResourceTypeDbPostgres:
+				err = scaffold.BindToPostgres(sourceType, userSpec, infraSpec.DbPostgres)
+			case ResourceTypeDbMySQL:
+				err = scaffold.BindToMySql(sourceType, userSpec, infraSpec.DbMySql)
+			case ResourceTypeDbMongo:
+				err = scaffold.BindToMongoDb(sourceType, userSpec, infraSpec.DbCosmosMongo)
+			case ResourceTypeDbCosmos:
+				err = scaffold.BindToCosmosDb(sourceType, userSpec, infraSpec.DbCosmos)
+			case ResourceTypeDbRedis:
+				err = scaffold.BindToRedis(sourceType, userSpec, infraSpec.DbRedis)
+			case ResourceTypeMessagingServiceBus:
+				err = scaffold.BindToServiceBus(sourceType, userSpec, infraSpec.AzureServiceBus)
+			case ResourceTypeMessagingKafka, ResourceTypeMessagingEventHubs:
+				err = scaffold.BindToEventHubs(sourceType, userSpec, infraSpec.AzureEventHubs)
+			case ResourceTypeStorage:
+				err = scaffold.BindToStorageAccount(sourceType, userSpec, infraSpec.AzureStorageAccount)
+			case ResourceTypeOpenAiModel:
+				err = scaffold.BindToAIModels(sourceType, userSpec, usedResource.Name)
+			case ResourceTypeHostContainerApp:
+				usedSpec := getServiceSpecByName(infraSpec, usedResource.Name)
+				if usedSpec == nil {
+					return fmt.Errorf("'%s' uses '%s', but %s doesn't exist", userSpec.Name, usedResource.Name,
+						usedResource.Name)
+				}
+				scaffold.BindToContainerApp(userSpec, usedSpec)
+			default:
+				return fmt.Errorf("resource (%s) uses (%s), but the type of (%s) is (%s), which is unsupported",
+					userResource.Name, usedResource.Name, usedResource.Name, usedResource.Type)
+			}
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func sourceType(projectConfig *ProjectConfig, userResourceName string) binding.SourceType {
+	userService := projectConfig.Services[userResourceName]
+	if userService == nil {
+		return binding.Unknown
+	}
+	switch userService.Language {
+	case ServiceLanguageJava:
+		return binding.Java
+	default:
+		return binding.Unknown
+	}
+}
+
+func printEnvListAboutUses(infraSpec *scaffold.InfraSpec, projectConfig *ProjectConfig,
+	console input.Console, ctx context.Context) error {
+	for i := range infraSpec.Services {
+		userSpec := &infraSpec.Services[i]
+		userResourceName := userSpec.Name
+		userResource, ok := projectConfig.Resources[userResourceName]
+		if !ok {
+			return fmt.Errorf("service (%s) exist, but there isn't a resource with that name",
+				userResourceName)
+		}
+		sourceType := sourceType(projectConfig, userResourceName)
+		for _, usedResourceName := range userResource.Uses {
+			usedResource, ok := projectConfig.Resources[usedResourceName]
+			if !ok {
+				return fmt.Errorf("in azure.yaml, (%s) uses (%s), but (%s) doesn't",
+					userResourceName, usedResourceName, usedResourceName)
+			}
+			console.Message(ctx, fmt.Sprintf("\nInformation about environment variables:\n"+
```

Comment:
There's some duplication between this and `add_preview.go`, which also prints out the environment variables